### PR TITLE
EUI-8715/EUI-8751: Register Other Org - regulatory organisation page functionality and "Remove" button (display only)

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
[EUI-8715](https://tools.hmcts.net/jira/browse/EUI-8715)
[EUI-8751](https://tools.hmcts.net/jira/browse/EUI-8751)

### Change description ###
EUI-8715: Implement regulatory organisation page functionality; EUI-8751: Add "Remove" button for regulator entries (display only).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM, actually I will change my component to make it consistent with this because I was not aware of xuilib-hmcts-error-summary (because of the name)

File: src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts [hunk 1]
```diff
@@ -121,12 +130,21 @@ export class RegulatoryOrganisationTypeComponent extends RegisterComponent imple
 
   public onContinueClicked(): void {
     if (this.validateForm()) {
-      this.registrationData.regulatoryOrganisationTypes = this.regulators.value as RegulatoryOrganisationType[];
+      // Set corresponding registration data
+      this.registrationData.regulators = this.regulators.value as Regulator[];
       // eslint-disable-next-line @typescript-eslint/no-empty-function
-      this.router.navigate(['/register-org-new/organisation-services-access']).then(() => {});
+      this.router.navigate(['/register-org-new/organisation-services-access']);
```
Comment (Daniel-Lam) [inline src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts:132]: @johnbenjamin-hmcts - `this.router.navigate()` returns a `Promise`, hence originally I added an empty function to resolve it. Are you sure this is no longer required? If it isn't, please delete the comment on Line 135

**Review by Daniel-Lam — COMMENTED** [review]
@johnbenjamin-hmcts - just added one observation

File: src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts (hunk 1 above)
Comment (DavidJayakumar) [inline src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts:131]: The new path should be constructed using service registerOrgService.REGISTER_ORG_NEW_ROUTE

**Review by DavidJayakumar — COMMENTED** [review]
A minor comment

File: src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/register-org/components/regulatory-organisation-type/regulatory-organisation-type.component.ts]: I have updated the code now.

**Review by DavidJayakumar — APPROVED** [review]
LGTM

**Review by DavidJayakumar — APPROVED** [review]
LGTM

**Pull request merged**